### PR TITLE
Ticket/aotech 6705 date based deltas

### PR DESCRIPTION
**Pull request by phatsk**
This change allows the user to set a date by which posts modified after are the only posts synced.

### CODE REVIEW

File: includes/class-dashboard.php [hunk 1]
```diff
@@ -141,14 +141,14 @@ public function register_settings() {
 		// @TODO update option names and locations below this line.
 		// Export page.
 		register_setting( 'press-sync-export', 'ps_options' );
-		register_setting( 'press-sync-export', 'ps_ignore_comments' );
```
Comment (jmichaelward) [inline includes/class-dashboard.php]: Why is this removed as part of this ticket?

File: includes/class-press-sync.php [hunk 2]
```diff
@@ -414,8 +424,9 @@ public function count_objects_to_sync( $objects_to_sync ) {
 
 		global $wpdb;
 
-		$where_clause = '';
-		$where_clause = $this->get_synced_object_clause( $objects_to_sync );
+		$where_clause  = '';
+		$where_clause  = $this->get_synced_object_clause( $objects_to_sync );
```
Comment (jmichaelward) [inline includes/class-press-sync.php:434]: This line just overwrites the previous line. Is the previous line needed?

File: views/dashboard/html-export.php [hunk 3]
```diff
@@ -68,4 +74,8 @@
 		<?php submit_button(); ?>
 	</form>
 </div>
-
+<script>
```
Comment (jmichaelward) [inline views/dashboard/html-export.php:78]: Nitpick: This is probably fine for now, but if we're on a custom admin screen we should really be enqueuing this JavaScript to load on that screen and not assigning it inline with our markup.

File: views/dashboard/html-export.php [hunk 4]
```diff
@@ -38,6 +38,12 @@
 					<span>This will <strong>only</strong> attempt to find and fix term/taxonomy relationships.</span>
 				</td>
 			</tr>
+			<tr valign="top">
+				<th scope="row">Sync Post Since Date</th>
```
Comment (rakeshdx) [inline views/dashboard/html-export.php:42]: We should probably specify the datetime as "EST" since we dont use unix timestamp or GMT time.

File: includes/class-press-sync.php [hunk 5]
```diff
@@ -46,6 +54,7 @@ class Press_Sync {
 	public function __construct() {
 		// Initialize plugin classes.
 		$this->plugin_classes();
+		$this->delta_date = date( 'Y-m-d 00:00:00', strtotime( get_option( 'ps_delta_date' ) ) ) ?: false;
```
Comment (jmichaelward) [inline includes/class-press-sync.php:57]: If it's already in the database we probably have bigger problems, but we're not sanitizing the ps_delta_date value here before we're using it and passing it to the date method. It'll probably make your date() method flunk out of class long before it gets appended as WHERE condition in your future SQL, but it's worth double-checking this to make sure there are no security implications here.

File: includes/class-press-sync.php [hunk 6]
```diff
@@ -1529,4 +1540,25 @@ public function maybe_get_terms_for_post( $where ) {
...
+	 *
+	 * @return string
+	 */
+	protected function get_posts_delta( $objects_to_sync ) {
+		if ( ! $this->delta_date ) {
+			return '';
+		}
+
+		// Currently only supported for posts.
+		if ( 'post' !== $objects_to_sync ) {
```
Comment (rakeshdx) [inline includes/class-press-sync.php:1564]: post meaning all different public post types? or just "post" post type

**Review by jmichaelward — CHANGES_REQUESTED** [review]
It's probably okay, but there are some direct database value usage issues I raised that need a second look, and a couple of other minor nitpicks.

File: includes/class-dashboard.php (hunk 1 above)
Comment (phatsk) [reply includes/class-dashboard.php]: It could easily be moved to its separate issue - I just noticed that that was where a different small bug was coming from. I'll remove it from this PR.

File: includes/class-press-sync.php (hunk 2 above)
Comment (phatsk) [reply includes/class-press-sync.php:434]: tbh I wasn't really paying attention to the lines here and above, it's just marked "changed" from indentation. Will note that for the future, however.

File: includes/class-press-sync.php (hunk 5 above)
Comment (phatsk) [reply includes/class-press-sync.php:57]: Considering `date` returns `false` on any non-numeric data, I don't think it's a huge concern, however, it's enough that I agree that sanitization for sanity's sake isn't a bad idea. I'll add that in.

File: includes/class-press-sync.php (hunk 6 above)
Comment (phatsk) [reply includes/class-press-sync.php:1564]: All different Custom Post Types etc (including Pages).

File: views/dashboard/html-export.php (hunk 4 above)
Comment (phatsk) [reply views/dashboard/html-export.php:42]: Good call, will add that.

File: views/dashboard/html-export.php (hunk 3 above)
Comment (phatsk) [reply views/dashboard/html-export.php:78]: Totally agree - actually just forgot to put it in the proper place, will update.

**Review by rakeshdx — APPROVED** [review]
